packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -1829,7 +2324,7 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin {
         final RouteSettings settings = route.settings;
         final Map<String, dynamic> settingsJsonable = <String, dynamic> {
           'name': settings.name,
-          'isInitialRoute': settings.isInitialRoute,
+          'isInitialRoute': settings.name == widget.initialRoute,
```

Comment:
This doesn't look correct. If "initialRoute" is "/a/b", then the router "/", "/a", and "/a/b" had isInitialRoute set to true. The changed version will only mark "/a/b" as initial route.

Do you know what this is used for?

Reply:
cc @devoncarew  who might know more about this.

Reply:
The intent of the previous version of this code was just to expose the `RouteSettings.isInitialRoute` field. From looking at the DevTools and IntelliJ plugin's code, we don't use this field. If RouteSettings no longer contains it, I think we can omit sending it.